extras/docker/fromsource/heketi-start.sh

```diff
@@ -74,4 +85,32 @@ if [[ -d "${BACKUPDB_PATH}" ]]; then
     fi
 fi
 
-/usr/bin/heketi --config=/etc/heketi/heketi.json
+# if NEED_TOPOLOGY_LOAD is not empty, start the service in the background, and
+# load the topology.json from the path in HEKETI_TOPOLOGY_FILE
+if [ -z "${NEED_TOPOLOGY_LOAD}" ]; then
+    # by default just start in the foreground
+    /usr/bin/heketi --config=/etc/heketi/heketi.json
+else
+    # start hketi in the background
+    /usr/bin/heketi --config=/etc/heketi/heketi.json &
+
+    # wait until heketi replies
+    while ! curl http://localhost:8080/hello; do
+        sleep 0.5
+    done
+
+    # load the topology
+    if [ -n "${HEKETI_ADMIN_KEY}" ]; then
+        HEKETI_SECRET_ARG="--secret='${HEKETI_ADMIN_KEY}'"
```

Comment:
Since this is bash, perhaps a bash array would be cleaner for the args.

Reply:
Uh, thats possible, but maybe a little less common to do in shell scripts.

Reply:
I think it would be cleaner, but I won't require it for merge.

Reply:
I did not change it to arrays, those are almost never used in bash scripts that I have seen.

Reply:
it seems you have not seen too many bash scripts ... ;-)